lib/snmp.js

```diff
@@ -676,21 +677,37 @@ Session.prototype.getAll = function (options, callback) {
         }
 
         self.sendMsg(pkt, options, function (err, varbinds) {
-            if (options.abortOnError && err) {
-                callback(err);
-            } else {
-                if (varbinds) {
-                    results = results.concat(varbinds);
-                }
-                if (c < options.oids.length) {
-                    getOne(c);
+            if (!deadlineExpired) {
+                if (options.abortOnError && err) {
+                    if (!deadlineExpired) {
+                        clearTimeout(deadlineTimer);
+                    }
+                    callback(err);
                 } else {
-                    callback(null, results);
+                    if (varbinds) {
+                        results = results.concat(varbinds);
+                    }
+                    if (c < options.oids.length) {
+                        getOne(c);
+                    } else {
+                        if (!deadlineExpired) {
+                            clearTimeout(deadlineTimer);
+                        }
+                        callback(null, results);
+                    }
                 }
             }
         });
     }
 
+    if (options.deadline) {
+      function deadlineExpiredEvent() {
```

Comment:
I'd prefer this as a `var deadlineExpiredEvent = function() { ...` instead. I'm sure this works as well, but that would look clearer to me and avoid me having to wonder what happens when it's hoisted etc.